tests/jballerina-unit-test/src/test/resources/test-src/types/never/never-type.bal

```diff
@@ -0,0 +1,71 @@
+import ballerina/lang.'xml;
+
+const ASSERTION_ERROR_REASON = "AssertionError";
+
+//------------ Testing a function with 'never' return type ---------
+
+function functionWithNeverReturnType() returns never {
+    string a = "hello";
+    if (a == "a") {
+        a = "b";
+    } else {
+        a = "c";
+    }
+}
+
+function testTypeOfNeverReturnTypedFunction() {
+    any|error expectedFunctionType = typedesc<function () returns (never)>;
+
+    typedesc <any|error> actualFunctionType = typeof functionWithNeverReturnType;
+    
+    if (actualFunctionType is typedesc<function () returns (never)>) {
+        return;
+    }
+    panic error(ASSERTION_ERROR_REASON,
+                message = "expected '" + expectedFunctionType.toString() + "', found '" + actualFunctionType.toString () + "'");
+}
+
+function testNeverReturnTypedFunctionCall() {
+    functionWithNeverReturnType();
+}
+
+//------------ Testing record type with 'never' field ---------
+
+type InclusiveRecord record {
+    int j;
+    never p?;
+}; 
+
+type ExclusiveRecord record {|
+    int j;
+    never p?;
+|}; 
+
+function testInclusiveRecord() {
+    InclusiveRecord inclusiveRecord = {j:0, "q":1};
+}
+
+function testExclusiveRecord() {
+    ExclusiveRecord exclusiveRecord = {j:0};
+}
+
+//------------- Testing XML<never> -----------------
+
+function testXMLWithNeverType() {
+    xml<never> x = <xml<never>> 'xml:concat();  //generates an exmpty XML sequence and assign it to XML<never>
+}
+
+
+//---------------Test 'never' types with 'union-type' descriptors
+function testNeverWithUnionType1() {
+    string|never j;
```

Comment:
Should this be allowed if we are not allowing `never x;`?

Reply:
But the spec says

> for anytype T, the type T|never is the same as T.

Reply:
That would be useful for cases like the following:
```ballerina
public function main() {
    int x = foo();
}

function foo() returns int|never {
    return 10;
}
```

Reply:
let's check with the others as well